src/Extensions/Hosting.Services/Internal/OmexStatefulServiceRunner.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Fabric;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Microsoft.ServiceFabric.Services.Runtime;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	/// <summary>
+	/// Class to manage stateful service fabric service startup
+	/// </summary>
+	internal sealed class OmexStatefulServiceRunner : IOmexServiceRunner
+	{
+		private readonly string m_applicationName;
+		private readonly ServiceContextAccessor<StatefulServiceContext> m_contextAccessor;
+		private IEnumerable<IListenerBuilder<StatefulServiceContext>> ListenerBuilders { get; }
```

Comment:
Why are these getters whereas the ones above aren't? They're all private so I don't understand the differentiation

Reply:
Fields are used only inside of the class and properties are used in the nested class, it's done to avoid the need for passing all the properties to it explicitly and better names, which is a bit of a hack :)

Reply:
I think that design makes sense. We usually use getters everywhere, but I prefer to keep the two separately personally, so I agree with your design.